src/main/java/com/deloitte/elrr/aggregator/utils/ExtensionsUtil.java

```diff
@@ -0,0 +1,134 @@
+package com.deloitte.elrr.aggregator.utils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.stereotype.Component;
+
+import com.deloitte.elrr.aggregator.rules.ExtensionsConstants;
+import com.deloitte.elrr.elrraggregator.exception.AggregatorException;
+import com.deloitte.elrr.entity.types.GoalType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yetanalytics.xapi.model.AbstractActor;
+import com.yetanalytics.xapi.model.Activity;
+import com.yetanalytics.xapi.model.Context;
+import com.yetanalytics.xapi.util.Mapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class ExtensionsUtil {
+
+    /**
+     * @param context
+     * @param returnObject
+     * @return extensionsMap or LocalDateTime
+     * @throws AggregatorException
+     */
+    public Object getExtensions(Context context, String returnObject)
```

Comment:
ahhhh okay, i see what you are doing here. the correct approach is to use a class type in that field, not a string. So for instance it would be Actor.class. 

What you have done here is hinge the extension key on the type being returned. Forgetting that the type is in string form, it also is not extensible at all because what if multiple extensions have the same type? I think they already do, so this creates confusion.

You need to refactor this in such a way that it has three args:

`getExtensionValue(Context context, String key, Class returnType)`

I promise once you do that it will all be much clearer and easier to work with.

Reply:
Talk to me tomorrow about how to use generic return types hinging on a parameter, because that is what you need here rather than `Object`